alchemy/src/aws/vpc.ts

```diff
@@ -0,0 +1,525 @@
+ *   cidrBlock: "172.16.0.0/16",
+ *   enableDnsHostnames: true,
+ *   enableDnsSupport: true,
+ *   name: "Custom VPC",
+ *   tags: {
+ *     Environment: "Production",
+ *     Project: "E-Commerce"
+ *   }
+ * });
+ *
+ * @example
+ * // Create a VPC with dedicated tenancy (isolated hardware):
+ * const dedicatedVpc = await Vpc("isolated-vpc", {
+ *   cidrBlock: "192.168.0.0/16",
+ *   instanceTenancy: "dedicated",
+ *   name: "Isolated VPC",
+ *   tags: {
+ *     Compliance: "PCI-DSS"
+ *   }
+ * });
+ */
+export const Vpc = Resource(
+  "aws::Vpc",
+  async function (
+    this: Context<Vpc>,
+    id: string,
+    props: VpcProps
+  ): Promise<Vpc> {
+    // Create EC2 client with region if specified
+    const client = new EC2Client({
+      region: props.region,
+    });
+
+    if (this.phase === "delete") {
+      try {
+        if (this.output?.id) {
+          // Handle multiple possible error types during deletion
+          await ignore(
+            ["VpcNotFound", "InvalidVpcID.NotFound", "InvalidVpcID.Malformed"],
+            () =>
+              client.send(
+                new DeleteVpcCommand({
+                  VpcId: this.output.id,
+                })
+              )
+          );
+        }
+      } catch (error) {
+        console.error("Error deleting VPC:", error);
+      }
+
+      return this.destroy();
+    } else {
+      try {
+        // Format tags for AWS API
+        const tagSpecifications: TagSpecification[] = [];
+        const tags: Tag[] = [];
+
+        if (props.name) {
+          tags.push({
+            Key: "Name",
+            Value: props.name,
+          });
+        }
+
+        if (props.tags) {
+          for (const [key, value] of Object.entries(props.tags)) {
+            tags.push({
+              Key: key,
+              Value: value,
+            });
+          }
+        }
+
+        if (tags.length > 0) {
+          tagSpecifications.push({
+            ResourceType: ResourceType.vpc,
+            Tags: tags,
+          });
+        }
+
+        let vpcId: string;
+        let existingVpc;
+
+        if (this.phase === "update" && this.output?.id) {
+          // VPC core properties like CIDR block can't be directly modified
+          // But we can modify attributes and tags
+          const describeResponse = await client.send(
+            new DescribeVpcsCommand({
+              VpcIds: [this.output.id],
+            })
+          );
+
+          if (!describeResponse.Vpcs || describeResponse.Vpcs.length === 0) {
+            throw new Error(`VPC with ID ${this.output.id} not found`);
+          }
+
+          existingVpc = describeResponse.Vpcs[0];
+          vpcId = this.output.id;
+
+          // Update modifiable attributes
+          // DNS Hostnames
+          if (props.enableDnsHostnames !== undefined) {
+            await modifyVpcDnsHostnames(
+              client,
+              vpcId,
+              props.enableDnsHostnames
+            );
+          }
```

Comment:
What if last time we deployed when it was `true`, and then swapped to `undefined`?

Reply:
```ts
// old props (last time we created/updated)
this.props.enableDnsHostnames !== props.enableDnsHostnames
```

Reply:
`this.props` (old props)
`props` (new props)